redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/Node.kt

```diff
@@ -158,20 +145,20 @@ public class Node(
    */
   public var measuredHeight: Int = -1
 
-  /**
-   * A callback to to measure this node according to the `widthSpec` and `heightSpec` constraints
-   * and update [measuredWidth] and [measuredHeight] with the result.
-   */
-  public var measure: (widthSpec: MeasureSpec, heightSpec: MeasureSpec) -> Size = { widthSpec, heightSpec ->
-    measuredWidth = MeasureSpec.resolveSize(width, widthSpec)
-    measuredHeight = MeasureSpec.resolveSize(height, heightSpec)
-    Size(measuredWidth, measuredHeight)
-  }
+  public var left: Int = -1
+  public var top: Int = -1
+  public var right: Int = -1
+  public var bottom: Int = -1
 
   /**
-   * A callback to place the node inside the given `left`, `top`, `right`, and `bottom` coordinates.
+   * Assign the node's position.
    */
-  public var layout: (left: Int, top: Int, right: Int, bottom: Int) -> Unit = { _, _, _, _ -> }
+  public fun layout(left: Int, top: Int, right: Int, bottom: Int) {
```

Comment:
Open question – who should own this state? It’s definitely an output of the layout engine and so there’s a simple case to be made that this function belongs on `FlexboxEngine.Adapter` instead.